pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_events.go

```diff
@@ -26,24 +26,19 @@ import (
 )
 
 func TestDuplicatedEventForUpgrade(events monitorapi.Intervals, kubeClientConfig *rest.Config) []*junitapi.JUnitTestCase {
-	allowedPatterns := []*regexp.Regexp{}
-	allowedPatterns = append(allowedPatterns, AllowedRepeatedEventPatterns...)
-	allowedPatterns = append(allowedPatterns, AllowedUpgradeRepeatedEventPatterns...)
+	// append upgrade allowances to the main slice
+	allowedDupeEvents := []*AllowedDupeEvent{}
+	allowedDupeEvents = append(allowedDupeEvents, AllowedRepeatedEvents...)
+	allowedDupeEvents = append(allowedDupeEvents, AllowedRepeatedUpgradeEvents...)
 
 	evaluator := duplicateEventsEvaluator{
-		allowedRepeatedEventPatterns: allowedPatterns,
-		allowedRepeatedEventFns:      AllowedRepeatedEventFns,
-		knownRepeatedEventsBugs:      KnownEventsBugs,
```

Comment:
This was wild, but virtually everything passed to the evaluator was never used. You'll notice these variables are never read from. Somewhere along the line the logic changed to "skip if flagged interesting", implying we don't need to check these anymore. However the code flagging interesting was *assuming* to check both upgrade and non-upgrade events, so effectively we were always allowing things we intended to only allow on upgrade.